sources/packages/web/src/components/generic/CheckAESTPermissionRole.vue

```diff
@@ -0,0 +1,39 @@
+<template>
+  <slot :isReadonly="isReadonly"></slot>
+</template>
+
+<script lang="ts">
+import { computed, PropType } from "vue";
+import { ClientIdType, Role } from "@/types";
+import { AuthService } from "@/services/AuthService";
+
+export default {
+  props: {
+    role: {
+      type: String as PropType<Role>,
+      required: true,
+    },
+  },
+  setup(props: any) {
+    const isReadonly = computed(() => {
+      const userToken = AuthService.shared.userToken;
+      if (userToken?.resource_access && userToken?.azp) {
+        if (userToken.azp === ClientIdType.AEST) {
+          const userRoles = userToken.resource_access[userToken.azp].roles;
+          return !userRoles?.includes(props.role);
+        }
+        /**
+         * Non aest client will come here. for eg, common components used in non AEST client like institution/student/supporting-user
```

Comment:
I would say that usually in the middle of the code we use more of the " //"  comment style.